proxy/proxy_test.go

```diff
@@ -81,7 +81,11 @@ func TestProxyHTTP(t *testing.T) {
 		Addr:    testProxyAddr,
 		Handler: http.HandlerFunc(p.ServeHTTP),
 	}
-	go func() { _ = server.ListenAndServe() }()
+	go func() {
+		if err := server.ListenAndServe(); err != http.ErrServerClosed {
+			t.Errorf("Error serving on %s: %v", testProxyAddr, err)
```

Comment:
Perhaps fail the test all together in this case?

Reply:
As of 1.16, go emits a warning when `t.Fatal` is called within a goroutine.

Reply:
Yes, you can't fail a test from a goroutine (or aren't supposed to, now enforced through the linter as @wpaulino pointed out). And it seemed a bit overkill to add an err chan to get the error out in this case. At least you see the error when running the test manually and know what's up.